engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -103,6 +103,8 @@ public boolean connect(Node ... nodeList) {
                 logger.warn("Trying to connect two already connected nodes, " + fromNode.getClass() + " and " + toNode.getClass());
             }
 
+            fromNode = toNode;
```

Comment:
Note that the logic was that every pair of adjecent nodes in the list becomes a `fromNode, toNode` and a connection is established between them, and after iterating through all pairs, the entire list is completed. However, I forgot the `mark current toNode as the next fromNode`, which lead to some.. "interesting" results.